In [2]:
#importing numpy and stats librayies
import numpy as np
import scipy.stats as stats

In [3]:
# Sample data
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

### Build 99% Confidence Interval Using Sample Standard Deviation
#### Using the t-distribution.

In [12]:
# Calculating the Sample mean and Sample standard deviation
sample_mean = round(np.mean(data),2)
sample_std = 0.5 # assuming 0.5 value
sample_size = len(data)
print("Sample_mean:",sample_mean,"\nSample_std:",sample_std,"\nSample_size:",sample_size)

Sample_mean: 1.24 
Sample_std: 0.5 
Sample_size: 15


In [9]:
# Confidence Level
confident_level = 0.99
alpha = round((1 - confident_level),2)
print("Alpha:",alpha)

Alpha: 0.01


In [10]:
# Calculating the critical value for a 99% confident interval
degree_freedom = sample_size -1
critical_value = stats.t.ppf(1 - alpha / 2, degree_freedom)
print("Degree of Freedom:",degree_freedom,"\nCritical_value:",critical_value)

Degree of Freedom: 14 
Critical_value: 2.97684273411266


In [14]:
# Calculate Standard Error 
standard_error = sample_std/np.sqrt(sample_size)
print("Standard_error:",standard_error)

Standard_error: 0.12909944487358055


In [15]:
# Calculate margin error
margin_error = round(critical_value * standard_error,2)
print("Margin_error:",margin_error)

Margin_error: 0.38


In [16]:
# Constructing Confidence Interval

confidence_interval = (sample_mean - margin_error, sample_mean + margin_error)
print("Confidenc_Interval:",confidence_interval)

Confidenc_Interval: (0.86, 1.62)


In [70]:
print(f"Sample Mean: {sample_mean}")
print(f"Sample Standard Deviation: {sample_std}")
print(f"Sample Size: {sample_size}")
print(f"Standard Error: {standard_error:.4f}")
print(f"Degrees of Freedom: {degree_freedom}")
print(f"Critical Value (t): {critical_value:.4f}")
print(f"Margin of Error: {margin_error:.4f}")
print(f"99% Confidence Interval: {confidence_interval}")

Sample Mean: 1.24
Sample Standard Deviation: 0.5
Sample Size: 15
Standard Error: 0.1291
Degrees of Freedom: 14
Critical Value (t): 2.9768
Margin of Error: 0.3800
99% Confidence Interval: (0.86, 1.62)


### Build 99% Confidence Interval Using Known Population Standard Deviation
#### Using the Z-distribution (normal distribution).

In [49]:
# Calculating the Sample mean and Sample standard deviation
sample_mean = round(np.mean(data),4)
sample_std1 = 0.2 # assuming 0.5 value
sample_size = len(data)
sample_mean,sample_std,sample_size

(1.2387, 0.5, 15)

In [58]:
# Confidence Level
confident_level = 0.99
alpha = 1 - confident_level

# Calculating the critical value for a 99% confident interval
degree_freedom = sample_size -1
critical_value = stats.t.ppf(1 - alpha / 2, degree_freedom)

# Calculate Standard Error 
standard_error = sample_std1/np.sqrt(sample_size)

# Calculate margin error
margin_error = round(critical_value * standard_error,4)

# Constructing Confidence Interval
confidence_interval = (sample_mean - margin_error, sample_mean + margin_error)

In [59]:
print(f"Sample Mean: {sample_mean}")
print(f"Sample Standard Deviation: {sample_std1}")
print(f"Sample Size: {sample_size}")
print(f"Standard Error: {standard_error:.4f}")
print(f"Degrees of Freedom: {degree_freedom}")
print(f"Critical Value (t): {critical_value:.4f}")
print(f"Margin of Error: {margin_error:.4f}")
print(f"99% Confidence Interval: {confidence_interval}")

Sample Mean: 1.2387
Sample Standard Deviation: 0.2
Sample Size: 15
Standard Error: 0.0516
Degrees of Freedom: 14
Critical Value (t): 2.9768
Margin of Error: 0.1537
99% Confidence Interval: (1.085, 1.3923999999999999)


#### A confidence interval provides a range within which we expect the true population parameter to lie,based on our sample data.
#### This range helps quantify the uncertainty around our sample estimate and supports better decision-making in data science.

### Summary
#### > Using Sample Standard Deviation (t-distribution):
#### -- Confidence Interval: (0.86, 1.62)
#### > Using Known Population Standard Deviation (z-distribution):
#### -- Confidence Interval: (1.085, 1.39239)